### ISSUE TITLE
Deep copy of Biggus Array

### ISSUE THREAD

**Issue by lbdreyer** [SciTools/biggus#157]
There is a problem with taking a deep copy of a Biggus Array.

It appears that doing so can change the order of the elements in the array.

We can define our own copy implementation by adding a `__deepcopy__` method. If we do so, the question is what should our implementation be?

**Comment by pelson** [comment]
> If we do so, the question is what should our implementation be?

Let's focus on defining a test which currently fails, then we can discuss exactly how the ```__deepcopy__``` implementation should look

**Comment by pelson** [comment]
The problem boils down to a numpy bug, which is impacting ArrayStack:

```
>>> a = np.array([[0, 1], [2, 3]], order='F', dtype='O')
>>> b = copy.deepcopy(a)

>>> print(a == b)
[[ True False]
 [False  True]]

>>> print(a, b)
array([[0, 1],
       [2, 3]], dtype=object)
array([[0, 2],
       [1, 3]], dtype=object)
```

Looks like we are going to have to put a guard into biggus for this. I'll raise the issue with numpy shortly.

**Comment by pelson** [comment]
I've submitted a fix to the problem in #159. We should test it out, and when happy, merge and create a new biggus release.
